src/service.ts

```diff
@@ -207,36 +209,46 @@ export class DebugService implements IDebugger {
     return reply.body.variables;
   };
 
-  getBreakpoints = (): DebugProtocol.SourceBreakpoint[] => {
-    return this._model.breakpointsModel.breakpoints.map(breakpoint => {
-      return {
-        line: breakpoint.line
-      };
-    });
-  };
-
   setBreakpoints = async (
     breakpoints: DebugProtocol.SourceBreakpoint[],
     path: string
   ) => {
     // Workaround: this should not be called before the session has started
     const client = this.session.client as IClientSession;
     await client.ready;
-    await this.session.sendRequest('setBreakpoints', {
+    return await this.session.sendRequest('setBreakpoints', {
       breakpoints: breakpoints,
       source: { path },
       sourceModified: false
     });
   };
 
-  updateBreakpoints = async () => {
+  updateBreakpoints = async (breakpoints: Breakpoints.IBreakpoint[]) => {
```

Comment:
I think that you need the same workaround as in `setBreakpoints` to avoid errors when switching to a new notebook for which the kernel is not ready.

Reply:
Maybe we can keep the same workaround for both methods, since `updateBreakpoints` uses `setBreakpoints` at the moment.

Reply:
The call to `setBreakpoints` sometimes returns an empty response. This is happening when starting a fresh new notebook and its kernel session was not started before.

It looks like the debug messages are intertwined (most likely because of https://github.com/jupyterlab/debugger/issues/101)

Reply:
This should be fixed with #130. No `setBreakpoints` request will be sent when starting a new notebook or switching to another one previously opened, only a restore session should be called, which will ask the backend for the breakpoints, update the model, and let the view update itself upon receiving signals.